packages/lesswrong/lib/collections/comments/callbacks.js

```diff
@@ -269,3 +269,44 @@ function NewCommentNeedsReview (comment) {
 addCallback("comments.new.async", NewCommentNeedsReview);
 
 addEditableCallbacks({collection: Comments, options: makeEditableOptions})
+
+async function validateDeleteOperations (modifier, comment, currentUser) {
+  if (modifier.$set) {
+    const { deleted, deletedPublic, deletedReason } = modifier.$set
+
+    if (deletedPublic && !deleted) {
+      throw new Error("You cannot publicly delete a comment without also deleting it")
+    }
+
+    if (deletedPublic && !deletedReason) {
+      throw new Error("Publicly deleted comments need to have a deletion reason");
+    } 
+
+    if (
+      (comment.deleted || comment.deletedPublic) && 
+      (deletedPublic || deletedReason) && 
+      !Users.canDo('comments.remove.all') && 
+      comment.deletedByUserId !== currentUser._id) {
+        throw new Error("You cannot edit the deleted status of a comment that's been deleted by someone else")
+    }
+
+    if (deletedReason && !deleted && !deletedPublic) {
+      throw new Error("You cannot set a deleted reason without deleting a comment")
+    }
+    
+    const childrenComments = await Comments.find({parentCommentId: comment._id})
```

Comment:
This `find` should be wrapped inside some sort of if statement; write now it applies to *all* `comments.edit.sync`, not just deletions.